### PR TITLE
docs: add explicit instruction for windows filepaths

### LINKED ISSUE #34125: terraform test: `-filter` flag not working on windows
### Terraform Version

```shell
Terraform v1.6.2
on windows_amd64
```

### Terraform Configuration Files

In `main.tf`:
```tf
variable "input_value" {
  type = string

  validation {
    condition = contains(["foo", "bar"], var.input_value)
    error_message = "not right!"
  }
}
```
In `tests/test_bad_input.tftest.hcl`:
```hcl
run "test_bad_input_wrong_value" {
  command = plan # don't need to apply for this test
  variables {
    input_value = "wrong value"
  }
  expect_failures = [ var.input_value ]
}
```

### Debug Output

```console
$ TF_LOG=trace terraform test -filter=test_bad_input.tftest.hcl
2023-10-23T13:07:05.015+0200 [INFO]  Terraform version: 1.6.2
2023-10-23T13:07:05.018+0200 [DEBUG] using github.com/hashicorp/go-tfe v1.36.0
2023-10-23T13:07:05.018+0200 [DEBUG] using github.com/hashicorp/hcl/v2 v2.19.1
2023-10-23T13:07:05.018+0200 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2023-10-23T13:07:05.019+0200 [DEBUG] using github.com/zclconf/go-cty v1.14.1
2023-10-23T13:07:05.019+0200 [INFO]  Go runtime version: go1.21.1
2023-10-23T13:07:05.019+0200 [INFO]  CLI args: []string{"REDACTED\\terraform_1.6.2_windows_amd64\\terraform.exe", "test", "-filter=test_bad_input.tftest.hcl"}
2023-10-23T13:07:05.020+0200 [TRACE] Stdout is a terminal of width 252
2023-10-23T13:07:05.020+0200 [TRACE] Stderr is a terminal of width 252
2023-10-23T13:07:05.021+0200 [TRACE] Stdin is a terminal
2023-10-23T13:07:05.032+0200 [DEBUG] Attempting to open CLI config file: REDACTED\AppData\Roaming\terraform.rc
2023-10-23T13:07:05.033+0200 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2023-10-23T13:07:05.040+0200 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2023-10-23T13:07:05.044+0200 [DEBUG] ignoring non-existing provider search directory REDACTED\AppData\Roaming\terraform.d\plugins
2023-10-23T13:07:05.055+0200 [DEBUG] ignoring non-existing provider search directory REDACTED\AppData\Roaming\HashiCorp\Terraform\plugins
2023-10-23T13:07:05.056+0200 [INFO]  CLI command args: []string{"test", "-filter=test_bad_input.tftest.hcl"}
2023-10-23T13:07:05.076+0200 [DEBUG] checking for provisioner in "."
2023-10-23T13:07:05.084+0200 [DEBUG] checking for provisioner in "REDACTED\\terraform_1.6.2_windows_amd64"
2023-10-23T13:07:05.086+0200 [DEBUG] TestSuiteRunner: found 0 files with 0 run blocks

Success! 0 passed, 0 failed.
```

### Expected Behavior

Tests are run

### Actual Behavior

No tests are run

### Steps to Reproduce

1. `terraform init`
2. `terraform test -filter=test_bad_input.tftest.hcl`

### Additional Context

image

### References

_No response_

### PR DESCRIPTION
**Pull request by liamcervante**
Fixes #34125 

## Target Release

1.6.4

### CODE REVIEW

File: website/docs/cli/commands/test.mdx [hunk 1]
```diff
@@ -55,3 +55,80 @@ The `-cloud-run` option accepts a [private registry module source](/terraform/la
...
+In addition, we have created a [setup module](/terraform/language/tests#modules) for our tests within the `testing` directory.
+
+In order to execute our tests we should run `terraform test` from the root configuration directory as if we were running `terraform plan` or `terraform apply`. Despite our actual test files being in the nested `tests` directory, we still operate Terraform from the main configuration.
+
+We could also execute specific test files using the `-filter` option.
+
+Linux, Mac OS, and UNIX:
+
+```shell
+terraform test -filter=tests/validations.tftest.hcl`
```
Comment (radditude) [inline website/docs/cli/commands/test.mdx:91]: I don't think it makes a difference to the rendered website, but the extra backtick at the end of this line appears to be breaking Github's syntax highlighting 😆

**Review by radditude — APPROVED** [review]
Makes sense to me content-wise, but using "we" to address the reader (as in "We could execute specific test files using `-filter`") made me 🤔 a bit since our documentation generally addresses the reader as "you" (as in "You can execute specific test files").

I'd be curious to get a second opinion from the technical writing team if that's an option!

**Comment by apparentlymart** [comment]
I am _not_ the technical writing team :grinning: but as a person who previously made similar mistakes I have learned that in our typical docs writing style:
- "you" means the reader/user, as in "You can ..." or "You must ..."
- "we" means some relevant party at HashiCorp (mainly just for sentences like "We recommend [some suggested practice]")
- we can also refer to "Terraform" as if it's an independent agent when describing the behavior of the product, as in "When [specific situation occurs], Terraform [does a thing]"
    - (This one can combine with the first rule, like "If/when you [do a specific thing], Terraform [reacts in some way]." In this case, prefer to describe it in present tense rather than future tense, i.e. don't say "Terraform _will_ [react in some way]".)

I am ready to be corrected if I've mistaken any of these but hopefully that's a reasonable approximation to start with. :grin: 

Some of our existing docs do not properly follow these conventions because some docs are pretty old by now, and so we also try to "fix forward" when working in the vicinity of some older docs. However, that doesn't seem relevant here since this page itself is relatively new and this new content is largely isolated from what's already there.

File: website/docs/cli/commands/test.mdx (hunk 1 above)
Comment (liamcervante) [reply website/docs/cli/commands/test.mdx]: Fixed! Was just a typo!

**Comment by liamcervante** [comment]
Thanks for the reviews! I've updated it following the guidelines mentioned by @apparentlymart. I think it's looking better now.

**Review by radditude — APPROVED** [review]
It looks so nice I had to approve it twice ✨

**Pull request merged**